components/common/accordion/accordion.tsx

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {RefObject} from 'react';
+
+import AccordionItem from './accordion_item';
+import './accordion.scss';
+
+export type AccordionDataType = {
+    title: string;
+    description?: string;
+    icon?: React.ReactNode;
+    items: React.ReactNode[];
+};
+
+type Props = {
+    items: AccordionDataType[];
+    openMultiple?: boolean;
+    onHeaderClick?: <T>(ref: RefObject<HTMLLIElement>) => T | void;
+};
+
+const Accordion = ({
+    items,
+    openMultiple,
```

Comment:
I'm a bit confused about the `openMultiple` prop. I don't see `<AccordionItem>` components having any collapse/expand states in the designs, but yet they're receiving the prop.

Is this a way to configure whether multiple `<Accordion>` components are allowed to be expanded at once? If so I would question whether a `<Accordion>` (let alone an `<AccordionItem>`) should have knowledge about the state of other sibling components, and would suggest perhaps an `<AccordionList>` that contains that logic.

I may be totally misunderstanding something though.

Reply:
@mkraft you are right about the `Accordion` component the siblings don't have any insight on the open/closed state of others though

Reply:
I don't see <AccordionItem> components having any collapse/expand states in the designs

> <img width="675" alt="Captura de pantalla 2022-01-21 a las 13 52 28" src="https://user-images.githubusercontent.com/10082627/150530103-62c138f8-7e05-4e71-802b-9eb269380283.png">
> hey @mkraft , so,  as shown above, left side is collapsed state, right side is expanded state.


Is this a way to configure whether multiple <Accordion> components are allowed to be expanded at once?

> Yes, this defines if only one item open at a time or multiple. This is covered in the testing for accordion too. I moved the control for this logic to the Accordion and the tests passed ok, since the test focus on the behavior and not in the internal implementation https://github.com/mattermost/mattermost-webapp/pull/9666/files#diff-5dc32f53bc2f17a06eb337f21adeca4ec4d8fc25c3548edb212899ca3679eec7R79

If so I would question whether a <Accordion> (let alone an <AccordionItem>) should have knowledge about the state of other sibling components, and would suggest perhaps an <AccordionList> that contains that logic

> Accordion now have the control over which AccordionItems are expanded/Collapsed